wgsl/index.bs

```diff
@@ -528,14 +528,22 @@ An <dfn>identifier</dfn> is a kind of [=token=] used as a name.
 See [[#declaration-and-scope]] and [[#directives]].
 
 The form of an identifier is defined via pattern-matching, except that
-an identifier must not have the same spelling as a [=keyword=] or as a [=reserved word=].
+an identifier:
+* must not have the same spelling as a [=keyword=] or as a [=reserved word=].
+* must not be `_` (a single underscore)
+* must not start with two undercores.
```

Comment:
We should promote this to the grammar.

Reply:
I had feared it would get unwieldy, but I did manage to fit reasonably well in one regexp.
Thanks for the suggestion.